src/Domains/Subscription/Subscriptions/Actions/CreateSubscription.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Kanvas\Subscription\Subscriptions\Actions;
+
+use Kanvas\Subscription\Subscriptions\Models\Subscription;
+use Kanvas\Subscription\Subscriptions\DataTransferObject\Subscription as SubscriptionDto;
+
+class CreateSubscription
+{
+    public function __construct(
+        protected SubscriptionDto $subscriptionDto
+    ) {
+    }
+
+    public function execute(): Subscription
+    {
+        $subscription = Subscription::firstOrcreate([
+            'users_id' => $this->subscriptionDto->user->getId(),
```

Comment:
@Keysie27 first or create should be base on company, app, stripe id no? not all the params